### PR TITLE
use Legolas, test serialization

### LINKED ISSUE #83: Use Legolas.jl to define `Package` type?
https://github.com/beacon-biosignals/Legolas.jl

This would involve small changes to the package definition (and a dependency on Legolas):
```julia
using Legolas: @schema, @version
@schema "package-analyzer.package" Package

@version PackageV1 begin
    name::String # name of the package
    uuid::UUID # uuid of the package
    repo::String # URL of the repository
    subdir::String # subdirectory of the package in the repo
    reachable::Bool # can the repository be cloned?
    docs::Bool # does it have documentation?
    runtests::Bool # does it have the test/runtests.jl file?
    github_actions::Bool # does it use GitHub Actions?
    travis::Bool # does it use Travis CI?
    appveyor::Bool # does it use AppVeyor?
    cirrus::Bool # does it use Cirrus CI?
    circle::Bool # does it use Circle CI?
    drone::Bool # does it use Drone CI?
    buildkite::Bool # does it use Buildkite?
    azure_pipelines::Bool # does it use Azure Pipelines?
    gitlab_pipeline::Bool # does it use Gitlab Pipeline?
    license_files::Vector{LicenseTableEltype} # a table of all possible license files
    licenses_in_project::Vector{String} # any licenses in the `license` key of the Project.toml
    lines_of_code::Vector{LoCTableEltype} # table of lines of code
    contributors::Vector{ContributionTableElType} # table of contributor data
    version::Union{VersionNumber, Nothing} # the version number, if a release was analyzed
    tree_hash::String # the tree hash of the code that was analyzed
end
```

That would define a PackageV1 struct that is pretty much the same as our existing one, with reasonable definitions for `hash`, `isequal`, `==`, keyword argument constructers, and Arrow serialization.

One advantage is that we could more easily add new fields in a backwards compatible way. If we added `foo` and declared it as `::Union{Missing,Foo}`, then if we deserialized an older `PackageV1` table from Arrow, it would populate that field as `missing` when we `Legolas.read` such a table. We also could provide a default instead.

Another is that we could communicate schema-breaking changes by declaring a new version of `Package`, i.e. `PackageV2`. We could keep the old definition around if we wanted, to be able to continue to deserialize old tables. And Legolas uses Arrow's metadata to record which version is serialized in table so it could redirect it to the correct struct to deserialize to.

I'm not sure there's a huge need for this, but it could be handy. We also could define e.g. `LicenseTableEltype` as a Legolas row instead of as a particular `NamedTuple`, which could be used for the same kind of schema migration functionality for those items.

### PR DESCRIPTION
**Pull request by ericphanson**
closes #83 

I haven't tested w/ github auth yet, since I need to regen a new token.

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -131,18 +131,24 @@ struct Package
...
     gitlab_pipeline::Bool # does it use Gitlab Pipeline?
-    license_files::Vector{@NamedTuple{license_filename::String, licenses_found::Vector{String}, license_file_percent_covered::Float64}} # a table of all possible license files
+    license_files::Vector{LicenseV1}} # a table of all possible license files
     licenses_in_project::Vector{String} # any licenses in the `license` key of the Project.toml
-    lines_of_code::Vector{@NamedTuple{directory::String, language::Symbol, sublanguage::Union{Nothing, Symbol}, files::Int, code::Int, comments::Int, blanks::Int}} # table of lines of code
-    contributors::Vector{@NamedTuple{login::Union{String,Missing}, id::Union{Int,Missing}, name::Union{String,Missing}, type::String, contributions::Int}} # table of contributor data
-    version::Union{VersionNumber, Nothing} # the version number, if a release was analyzed
+    lines_of_code::Vector{LinesOfCodeV1} # table of lines of code
+    contributors::Vector{ContributorsV1} # table of contributor data
+    version::Union{String, Missing} # the version number, if a release was analyzed
```
Comment (ericphanson) [inline docs/src/index.md:138]: This is really annoying, but I just couldn't get `VersionNumber` working. The string has all the info we need though, so I guess it's OK. Xref https://github.com/apache/arrow-julia/issues/461

File: Project.toml [hunk 2]
```diff
@@ -1,14 +1,16 @@
 name = "PackageAnalyzer"
 uuid = "e713c705-17e4-4cec-abe0-95bf5bf3e10c"
 authors = ["Mosè Giordano <mose@gnu.org>"]
-version = "1.0.0"
+version = "2.0.0"
 
 [deps]
+Arrow = "69666777-d1a9-59fb-9406-91d4454c9d45"
```
Comment (giordano) [inline Project.toml:7]: Is Arrow used anywhere in the package?

Comment (ericphanson) [reply Project.toml:7]: Oops, didn’t mean to add it

Comment (ericphanson) [reply Project.toml:7]:
```suggestion
```

**Review by giordano — APPROVED** [review]
Looks good as far as I can empirically tell, thanks!

File: src/PackageAnalyzer.jl [hunk 3]
```diff
@@ -11,6 +11,8 @@ using Git: Git
 using Downloads
 using Tar
 using CodecZlib
+using Legolas
```
Comment (ericphanson) [inline src/PackageAnalyzer.jl:14]: We need Legolas in scope due to https://github.com/beacon-biosignals/Legolas.jl/issues/91

**Comment by ericphanson** [comment]
Should I go ahead and register this as v2, or do we want to hold off?

**Comment by giordano** [comment]
I'm happy with releasing a new version, versions are cheap and I don't expect we have many users

**Pull request merged**